### PR TITLE
Logout

### PR DESCRIPTION
**Pull request by antoniotrkdz**
Logout link in nav component, now connected.
Please merge Form multiselect branch first.
Relates to #92

### CODE REVIEW

File: react-ui/src/components/Form_Post_Job.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ import * as actions from '../actions/post_job.js';
 import {connect} from 'react-redux';
 import {Router, Route, IndexRoute, browserHistory} from 'react-router';
 import isEmpty from 'lodash/isEmpty';
+import Categories from '../constants/job_categories.js';
```
Comment (des-des) [inline react-ui/src/components/Form_Post_Job.js]: lower case for categories pls

File: react-ui/src/components/Form_Post_Job.js [hunk 2]
```diff
@@ -143,9 +144,19 @@ class Form_Post_Job extends Component {
     }
 
     render() {
+        const options = Categories.map(function(elem) {
```
Comment (des-des) [inline react-ui/src/components/Form_Post_Job.js]: lower case

File: react-ui/src/components/Form_Post_Job.js [hunk 3]
```diff
@@ -143,9 +144,19 @@ class Form_Post_Job extends Component {
...
+                    {elem}
+                </option>
+            );
+        });
+        
         if (!this.state) {
             return <div>Loading</div>;
         }
+        
+        console.log(this.state);
```
Comment (des-des) [inline react-ui/src/components/Form_Post_Job.js]: :scream:

File: react-ui/src/components/Form_Register_Student.js [hunk 4]
```diff
@@ -1,16 +1,20 @@
...
         super();
         this.onSubmit = this.onSubmit.bind(this);
         this.onChange = this.onChange.bind(this);
+        this.onChangeJobCat = this.onChangeJobCat.bind(this);
         this.checkEmail = this.checkEmail.bind(this);
 
         this.state = {
-            student: {},
+            student: {
+                job_cat: {},
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: oh no job cat still here

File: react-ui/src/components/Form_Register_Student.js [hunk 5]
```diff
@@ -35,7 +39,7 @@ class Form_Register_Student extends Component {
             //browserHistory.push('/dashboard');
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: :fire:

File: react-ui/src/components/Form_Register_Student.js [hunk 6]
```diff
@@ -46,7 +50,32 @@ class Form_Register_Student extends Component {
         });
     }
 
+    onChangeJobCat(evt) {
+        var student = this.state.student;
+        var job_cat = student.job_cat;
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: no snake case pls

File: react-ui/src/components/Form_Register_Student.js [hunk 7]
```diff
@@ -46,7 +50,32 @@ class Form_Register_Student extends Component {
...
+        this.setState({
+            student: {
+                ...student,
+                job_cat: [...job_cat, evt.target.value],
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: no cat pls

Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: no snake case pls

File: react-ui/src/components/Form_Register_Student.js [hunk 8]
```diff
@@ -46,7 +50,32 @@ class Form_Register_Student extends Component {
...
+            },
+        });
+    }
+
     render() {
+        const options = Categories.map(function(elem) {
+            return (
+                <option value={Categories[elem]}>
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: capitalisation

File: react-ui/src/components/Form_Register_Student.js [hunk 9]
```diff
@@ -46,7 +50,32 @@ class Form_Register_Student extends Component {
...
+                    {elem}
+                </option>
+            );
+        });
+
+        if (!this.state) {
+            return <div>Loading</div>;
+        }
+        
+        console.log(options);
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: :fire:

File: react-ui/src/components/Form_Register_Student.js [hunk 10]
```diff
@@ -46,7 +50,32 @@ class Form_Register_Student extends Component {
...
+        console.log(this.state);
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: :fire:

File: react-ui/src/components/Form_Register_Student.js [hunk 11]
```diff
@@ -224,8 +253,8 @@ class Form_Register_Student extends Component {
                         id="jobCategory1"
                         type="text"
                         placeholder="Select"
-                        value={this.state.student.index1}
-                        onChange={this.onChange}
+                        value={this.state.student.job_cat.index_1}
```
Comment (des-des) [inline react-ui/src/components/Form_Register_Student.js]: pls no snake case

File: react-ui/src/components/Nav.js [hunk 12]
```diff
@@ -1,54 +1,72 @@
...
-import {logout} from '../actions/AppActions';
+import {connect} from 'react-redux';
+import {logout} from '../actions/logout.js';
 import LoadingButton from './LoadingButton.js';
 
 class Nav extends Component {
     render() {
         // Render either the Log In and register buttons, or the logout button
         // based on the current authentication state.
+        console.log(this.state)
```
Comment (des-des) [inline react-ui/src/components/Nav.js]: please no

File: react-ui/src/components/Nav.js [hunk 13]
```diff
@@ -1,54 +1,72 @@
...
                     {navButtons}
                 </div>
             </div>
         );
     }
 
-    _logout() {
+    logout() {
+        var state = this.state;
+        this.setState({
```
Comment (des-des) [inline react-ui/src/components/Nav.js]: No, this state should be in redux

File: react-ui/src/components/Nav.js [hunk 14]
```diff
@@ -1,54 +1,72 @@
...
+            },
+        });
         this.props.dispatch(logout());
     }
 }
 
-export default Nav;
+function mapStateToProps(state) {
+    return {
+        logout: state.logout,
```
Comment (des-des) [inline react-ui/src/components/Nav.js]: bad name, logout sounds like a function.

This should be hold session data, ie, isLoggedIn, userName userType

File: react-ui/src/components/Nav.js [hunk 15]
```diff
@@ -1,54 +1,72 @@
...
+    };
+}
+
+export default connect(mapStateToProps, {logout})(Nav);
```
Comment (des-des) [inline react-ui/src/components/Nav.js]: okay weird shit happening here.

You need an actions file + reducer for this functionality

File: react-ui/src/components/pages/SearchStudents.js [hunk 16]
```diff
@@ -2,6 +2,7 @@ import React, {Component} from 'react';
 import {connect} from 'react-redux';
 import * as actions from '../../actions/search_students.js';
 import {Link} from 'react-router';
+import Categories from '../../constants/job_categories.js';
```
Comment (des-des) [inline react-ui/src/components/pages/SearchStudents.js]: capitalisation

File: react-ui/src/constants/job_categories.js [hunk 17]
```diff
@@ -0,0 +1,14 @@
+export default [
```
Comment (des-des) [inline react-ui/src/constants/job_categories.js]: I am pretty sure I have already look at this? ..

File: react-ui/src/reducers/logout_reducer.js [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+
+export default (state = initialState, action = {}) => {
+    switch (action.type) {
+    case REMOVE_AUTH:
+        return {
+            status: action.status,
+            error: action.error,
+            response: action.response,
+            state: initialState,
+            //loggedIn: false,
```
Comment (des-des) [inline react-ui/src/reducers/logout_reducer.js]: no

File: server/index.js [hunk 19]
```diff
@@ -267,6 +267,20 @@ server.register(plugins, err => {
         },
     });
 
+    server.route({
+        method: 'GET',
+        path: '/api/logout',
+        handler: (request, reply) => {
+            console.log(request);
```
Comment (des-des) [inline server/index.js]: :-1:

Comment (antoniotrkdz) [inline server/index.js]: Done.

File: react-ui/src/components/Form_Post_Job.js (hunk 1 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Post_Job.js]: they are gone afer merging in master.

File: react-ui/src/components/Form_Post_Job.js (hunk 2 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Post_Job.js]: as above

File: react-ui/src/components/Form_Post_Job.js (hunk 3 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Post_Job.js]: Done

File: react-ui/src/components/Form_Register_Student.js (hunk 4 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: as above

File: react-ui/src/components/Form_Register_Student.js (hunk 6 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: as above.

File: react-ui/src/components/Form_Register_Student.js (hunk 7 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: as above.

File: react-ui/src/components/Form_Register_Student.js (hunk 8 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: as above.

File: react-ui/src/components/Form_Register_Student.js (hunk 9 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: as above.

File: react-ui/src/components/Form_Register_Student.js (hunk 10 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: as above.

File: react-ui/src/components/Form_Register_Student.js (hunk 11 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Form_Register_Student.js]: as above

File: react-ui/src/components/Nav.js (hunk 12 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Nav.js]: gone.

File: react-ui/src/components/pages/SearchStudents.js (hunk 16 above)
Comment (antoniotrkdz) [inline react-ui/src/components/pages/SearchStudents.js]: as above

File: react-ui/src/constants/job_categories.js (hunk 17 above)
Comment (antoniotrkdz) [inline react-ui/src/constants/job_categories.js]: Maybe in the form multiselect

File: react-ui/src/components/Nav.js (hunk 13 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Nav.js]: Done.

File: react-ui/src/components/Nav.js (hunk 14 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Nav.js]: changed to isLogged

File: react-ui/src/components/Nav.js (hunk 15 above)
Comment (antoniotrkdz) [inline react-ui/src/components/Nav.js]: Done, I think

File: react-ui/src/actions/login.js [hunk 20]
```diff
@@ -10,10 +10,11 @@ export const loginRequest = (email, password) => dispatch => {
     axios
         .post('/api/auth', {email, password})
         .then(response => {
+            console.log('rrrr',response);
```
Comment (des-des) [inline react-ui/src/actions/login.js]: for why?

**Comment by des-des** [comment]
@antoniotrkdz conflicts

File: react-ui/src/actions/login.js (hunk 20 above)
Comment (antoniotrkdz) [inline react-ui/src/actions/login.js]: Removed.

**Comment by antoniotrkdz** [comment]
Thanks, I resolved the conflicts.

**Pull request merged**